core/src/main/java/org/apache/iceberg/CachingCatalog.java

```diff
@@ -126,20 +128,36 @@ public Transaction newReplaceTableTransaction(TableIdentifier ident, Schema sche
     // when the transaction commits, invalidate the table in the cache if it is present.
     return CommitCallbackTransaction.addCallback(
         catalog.newReplaceTableTransaction(ident, schema, spec, location, properties, orCreate),
-        () -> tableCache.invalidate(canonicalizeIdentifier(ident)));
+        () -> invalidate(canonicalizeIdentifier(ident)));
   }
 
   @Override
   public boolean dropTable(TableIdentifier ident, boolean purge) {
     boolean dropped = catalog.dropTable(ident, purge);
-    tableCache.invalidate(canonicalizeIdentifier(ident));
+    invalidate(canonicalizeIdentifier(ident));
     return dropped;
   }
 
   @Override
   public void renameTable(TableIdentifier from, TableIdentifier to) {
     catalog.renameTable(from, to);
```

Comment:
If we alter table properties (to change metadata partitioning for example) do we want to invalidate in that case as well?

Reply:
I think that is already taken care of since `CachingCatalog` forces the same `TableOperations` object to be used for metadata and main tables.

Reply:
There is another test in `TestCachingCatalog` for that.

Reply:
I think @HeartSaVioR already fixed that.